### PR TITLE
remove metadata start with underline for resource legacy output

### PR DESCRIPTION
**Pull request by 928PJY**

### CODE REVIEW

File: src/docfx/build/legacy/files/LegacyResource.cs [hunk 1]
```diff
@@ -19,11 +19,23 @@ internal static class LegacyResource
             var legacyManifestOutput = legacyManifestItem.Output;
             var metadata = metadataProvider.GetMetadata(doc);
             metadata = LegacyMetadata.GenerataCommonMetadata(metadata, docset);
+
+            var metadataNeedToBeRemove = new List<string> { "__global" };
+            foreach (var property in metadata)
```
Comment (live1206) [inline src/docfx/build/legacy/files/LegacyResource.cs]: use `for` loop to remove and iterate at the same time?

File: src/docfx/build/legacy/manifest/LegacyManifestItem.cs [hunk 2]
```diff
@@ -1,6 +1,7 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Collections.Generic;
```
Comment (live1206) [inline src/docfx/build/legacy/manifest/LegacyManifestItem.cs]: not in use

File: src/docfx/build/legacy/files/LegacyResource.cs (hunk 1 above)
Comment (live1206) [reply src/docfx/build/legacy/files/LegacyResource.cs]: not related

Comment (928PJY) [reply src/docfx/build/legacy/files/LegacyResource.cs]: cannot access jobject key by index

---
In reply to: [240453204](https://github.com/dotnet/docfx/pull/3751#discussion_r240453204) [](ancestors = 240453204)

**Pull request merged**